src/basic/ml/FStar_Util.ml

```diff
@@ -645,3 +645,29 @@ let basename = Filename.basename
 let print_endline = print_endline
 
 let map_option f opt = BatOption.map f opt
+
+let format_value_file_name (prefix:string) =
+  (* we use different suffixes for F# and OCaml because they use incompatible encodings of values. *)
+  format1 "%s.mlval" prefix
+
+let save_value_to_file (fname:string) value =
+  BatFile.with_file_out 
+    fname
+    (fun f ->
+      BatPervasives.output_value f value)
+
```

Comment:
confirmed. `FStar.main` wraps everything it calls within a `try...with | e ->`. 

in this case, however, i anticipate a `"Unexpected error; please file a bug report, ideally with a minimized version..."` message because this involves a standard library exception, not an exception defined in `F_Util` or `U_Util`.